libs/blocks/hero-marquee/hero-marquee.js

```diff
@@ -0,0 +1,208 @@
+    if (!elTaps.length) return;
+    const parts = elClass.split('-');
+    el.classList.add(`${parts[1]}-${parts[0]}`);
+    el.classList.remove(elClass);
+  });
+}
+
+function decorateList(el, classes) {
+  el.classList.add('body-l');
+  const listItems = el.querySelectorAll('ul li', 'ol li');
+  if (listItems.length) {
+    const firstDiv = el.querySelector(':scope > div');
+    firstDiv.classList.add('foreground');
+    [...listItems].forEach((item) => {
+      const firstElemIsIcon = item.children[0]?.classList.contains('icon');
+      if (firstElemIsIcon) item.classList.add('icon-item');
+      if (!item.parentElement.classList.contains('icon-list')) item.parentElement.classList.add('icon-list');
+    });
+  }
+  distillClasses(el, classes);
+}
+
+function decorateQr(el) {
+  const text = el.querySelector(':scope > div');
+  /* c8 ignore next */
+  if (!text) return;
+  const classes = ['qr-code-img', 'google-play', 'app-store'];
+  [...text.children].forEach((e, i) => {
+    e.classList.add(classes[i]);
+  });
+}
+
+function decorateLockupFromContent(el) {
+  const rows = el.querySelectorAll(':scope > p');
+  const firstRowImg = rows[0]?.querySelector('img');
+  if (firstRowImg) rows[0].classList.add('lockup-area');
+}
+
+function decorateBg(el) {
+  const block = el.closest('.hero-marquee');
+  block.style.background = el.textContent.trim();
+  el.remove();
+}
+
+function decorateText(el, classes) {
+  el.classList.add('norm');
+  const btnClass = classes?.find((c) => c.endsWith('-button'));
+  if (btnClass) {
+    const [theme, size] = btnClass.split('-').reverse();
+    el.classList.remove(btnClass);
+    decorateButtons(el, `${size}-${theme}`);
+  } else {
+    decorateButtons(el, 'button-xl');
+  }
+  distillClasses(el, classes);
+}
+
+function decorateLockupRow(el) {
+  const child = el.querySelector(':scope > div');
+  if (child) child.classList.add('lockup-area');
+}
+
+function decorateSup(el, classes) {
+  el.classList.add('norm');
+  distillClasses(el, classes);
+}
+
+function extendButtonsClass(copy) {
+  const buttons = copy.querySelectorAll('.con-button');
+  if (buttons.length === 0) return;
+  buttons.forEach((button) => {
+    button.classList.add('button-xl', 'button-justified-mobile');
+  });
+}
+function parseKeyString(str) {
+  const regex = /^(\w+)\s*\((.*)\)$/;
+  const match = str.match(regex);
+  if (!match) return { key: str };
+  const key = match[1];
+  const classes = match[2].split(',').map((c) => c.trim());
+  const result = { key, classes };
+  return result;
+}
+
+function loadContentType(el, key, classes) {
+  if (classes !== undefined && classes.length) el.classList.add(...classes);
+  if (key === 'bgcolor') decorateBg(el);
+  if (key === 'list') decorateList(el, classes);
+  if (key === 'lockup') decorateLockupRow(el);
+  if (key === 'qrcode') decorateQr(el);
+  if (key === 'supplemental') decorateSup(el, classes);
+  if (key === 'text') decorateText(el, classes);
```

Comment:
Was `switch` avoided intentionally? Looks like a good use-case for it

Reply:
No, not intention, it just ended up being more concise this way. 
I've updated this to use a `switch` instead which is likely more performant, but does increase line count.